### PR TITLE
[receiver/kubeletstats] Add new CPU utilization metrics

### LINKED ISSUE #24905: [receiver/kubeletstats] Add metric to represent the percentage of cpu and memory utilization compared to the configured limit
### Component(s)

_No response_

### Is your feature request related to a problem? Please describe.

The receiver reports the current memory and cpu usage for pods and containers, but unless you know the limits set for the containers in the pod it is hard to determine if that value is getting too high.

### Describe the solution you'd like

A new set of pod and container metrics that represents the utilization of memory and cpu that the pod/container is consuming based on the usage value and the configured limits.

- k8s.pod.memory.usagePercent
- k8s.pod.cpu.usagePercent
- k8s.container.memory.usagePercent
- k8s.container.cpu.usagePercent

### PR DESCRIPTION
**Pull request by TylerHelmuth**
**Description:** 
Adds new CPU utilization metrics with respect to pod/container CPU limits and requests 

**Link to tracking Issue:** <Issue number if applicable>
Closes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/24905

**Testing:** <Describe what testing was performed and which tests were added.>
Added new unit tests and tested locally

### CODE REVIEW

**Comment by TylerHelmuth** [comment]
Because of some bad changelog stuff from https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/25894 I added the Skip Changelog label to get it to pass, but we do have a changelog here.

**Comment by TylerHelmuth** [comment]
@dmitryax ready for review

**Review by evan-bradley — APPROVED** [review]

File: receiver/kubeletstatsreceiver/testdata/pods.json [hunk 1]
```diff
@@ -11,11 +11,11 @@
             "name":"kube-scheduler",
             "resources": {
               "requests": {
-                "cpu": "50m",
+                "cpu": "2m",
                 "memory": "10M"
               },
               "limits": {
-                "cpu": "100m",
+                "cpu": "4m",
```
Comment (dmitryax) [inline receiver/kubeletstatsreceiver/testdata/pods.json:18]: Why is this changed?

**Review by dmitryax — APPROVED** [review]
LGTM

File: receiver/kubeletstatsreceiver/testdata/pods.json (hunk 1 above)
Comment (TylerHelmuth) [reply receiver/kubeletstatsreceiver/testdata/pods.json:18]: I updated the test data to give us more interesting percentages to work with.  The previous values produced really small numbers when used with the values set for the first pod in https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/kubeletstatsreceiver/testdata/stats-summary.json

Comment (dmitryax) [reply receiver/kubeletstatsreceiver/testdata/pods.json:18]: > I updated the test data to give us more interesting percentages to work with. 

I'm just having troubles finding where this change is reflected. Is there a test output?

Comment (TylerHelmuth) [reply receiver/kubeletstatsreceiver/testdata/pods.json:18]: In the `scraper_test.go` file, these changes let me test that there was a value above 1 for `k8s.pod.cpu_request_utilization` and `k8s.container.cpu_request_utilization` but less than 1 for `k8s.pod.cpu_limit_utilization` and `k8s.container.cpu_limit_utilization`.

Comment (dmitryax) [reply receiver/kubeletstatsreceiver/testdata/pods.json:18]: I see. Thanks for clarifying. Given that `fakeRestClient` provides static values, I think we should generate pdata output and use it for validation the way we do it in most oh other tests. Maybe worth filing an issue with `good first issue` label for this.

Comment (TylerHelmuth) [reply receiver/kubeletstatsreceiver/testdata/pods.json:18]: Ya that is a good idea. I'll open an issue

Comment (TylerHelmuth) [reply receiver/kubeletstatsreceiver/testdata/pods.json:18]: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27468
